claims/parse_component_token.go

```diff
@@ -0,0 +1,45 @@
+// Copyright © 2016 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+package claims
+
+import (
+	"fmt"
+
+	"github.com/TheThingsNetwork/go-account-lib/tokenkey"
+)
+
+const (
+	RouterType  = "router"
+	BrokerType  = "broker"
+	HandlerType = "handler"
+)
+
+// FromComponentToken uses the tokenkey provider to parse and validate a token into its
+// corresponding claims
+func FromComponentToken(provider tokenkey.Provider, accessToken string) (*ComponentClaims, error) {
+	claims := &ComponentClaims{}
+
+	err := fromToken(provider, accessToken, claims)
+	if err != nil {
+		return nil, err
+	}
+
+	switch claims.Type {
+	case RouterType:
```

Comment:
If it's just a fallthrough, just do `case RouterType, BrokerType, HandlerType:`
If you plan to add more stuff later, this is ok